src/python/pants/option/subsystem.py

```diff
@@ -127,6 +128,28 @@ def __eq__(self, other: Any) -> bool:
         return bool(self.options == other.options)
 
 
+class AnalysisGoalToolSubsystem(Subsystem):
+    """Subsystem base for tools that invoked as part of one of the "analysis" goals.
+
+    E.g. fmt/lint/check
+    """
+
+    example_goal_name: ClassVar[str]
+    args_extra_help: ClassVar[str] = ""
+
+    skip = BoolOption(
+        default=False,
+        help=lambda subsystem_cls: (
+            f"If true, don't use {subsystem_cls.name} when running goals like "
+            + f" `{bin_name()} {subsystem_cls.example_goal_name}`"
+        ),
+    )
```

Comment:
Couldn't easily use `SkipOption` here, and the only reason it didn't get removed in this PR is because of `BufSubsystem` breaking the mold.

The help message slightly tweaked as well to be less "absolute" about which goal a tool might appear in. Plugins exist so pretending we know all the goals is folly.